guvnor-structure/guvnor-structure-client/src/main/java/org/guvnor/structure/client/editors/repository/fork/ForkRepositoryPresenter.java

```diff
@@ -0,0 +1,203 @@
+ * limitations under the License.
+ */
+
+package org.guvnor.structure.client.editors.repository.fork;
+
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.Dependent;
+import javax.inject.Inject;
+
+import org.guvnor.structure.client.editors.repository.RepositoryPreferences;
+import org.guvnor.structure.organizationalunit.OrganizationalUnit;
+import org.guvnor.structure.organizationalunit.OrganizationalUnitService;
+import org.guvnor.structure.repositories.Repository;
+import org.guvnor.structure.repositories.RepositoryAlreadyExistsException;
+import org.guvnor.structure.repositories.RepositoryEnvironmentConfigurations;
+import org.guvnor.structure.repositories.RepositoryService;
+import org.jboss.errai.bus.client.api.messaging.Message;
+import org.jboss.errai.common.client.api.Caller;
+import org.jboss.errai.common.client.api.ErrorCallback;
+import org.jboss.errai.common.client.api.RemoteCallback;
+import org.uberfire.client.mvp.PlaceManager;
+import org.uberfire.mvp.impl.DefaultPlaceRequest;
+import org.uberfire.rpc.SessionInfo;
+import org.uberfire.security.authz.AuthorizationManager;
+
+import static org.guvnor.structure.security.RepositoryFeatures.*;
+
+@Dependent
+public class ForkRepositoryPresenter implements ForkRepositoryView.Presenter {
+
+    private RepositoryPreferences repositoryPreferences;
+
+    private ForkRepositoryView view;
+
+    private Caller<RepositoryService> repositoryService;
+
+    private Caller<OrganizationalUnitService> organizationalUnitService;
+
+    private PlaceManager placeManager;
+
+    private AuthorizationManager authorizationManager;
+
+    private Map<String, OrganizationalUnit> availableOrganizationalUnits = new HashMap<String, OrganizationalUnit>();
+
+    private SessionInfo sessionInfo;
+    private boolean assetsManagementIsGranted = false;
+
+    @Inject
+    public ForkRepositoryPresenter( final RepositoryPreferences repositoryPreferences,
+                                    final ForkRepositoryView view,
+                                    final Caller<RepositoryService> repositoryService,
+                                    final Caller<OrganizationalUnitService> organizationalUnitService,
+                                    final PlaceManager placeManager,
+                                    final AuthorizationManager authorizationManager,
+                                    final SessionInfo sessionInfo ) {
+        this.repositoryPreferences = repositoryPreferences;
+        this.view = view;
+        this.repositoryService = repositoryService;
+        this.organizationalUnitService = organizationalUnitService;
+        this.placeManager = placeManager;
+        this.authorizationManager = authorizationManager;
+        this.sessionInfo = sessionInfo;
+    }
+
+    @PostConstruct
+    public void init() {
+        view.init( this,
+                   isOuMandatory() );
+        setAssetsManagementGrant();
```

Comment:
Please see [here](https://github.com/droolsjbpm/guvnor/commit/083845d8bdda17ff002544346af029c21ec6e386). The call to ```setAssetsManagementGrant()``` was moved to (in your case) ```showForm()``` to ensure ```KieACL``` has had time to load and initialise.